x-pack/plugins/ml/server/models/model_management/models_provider.ts

```diff
@@ -597,11 +598,26 @@ export class ModelsProvider {
     taskType: InferenceTaskType,
     modelConfig: InferenceModelConfig
   ) {
-    return await this._client.asCurrentUser.inference.putModel({
-      inference_id: inferenceId,
-      task_type: taskType,
-      model_config: modelConfig,
-    });
+    try {
+      const result = await this._client.asCurrentUser.inference.putModel(
+        {
+          inference_id: inferenceId,
+          task_type: taskType,
+          model_config: modelConfig,
+        },
+        { maxRetries: 0 }
+      );
+      return result;
+    } catch (error) {
+      // Request timeouts will usually occur when the model is being downloaded/deployed
```

Comment:
Shall we request a downloading tasks in that case to verify? 
You can try [this](https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/server/models/model_management/models_provider.ts#L607) method to retrieve a task id

Reply:
Mostly the team building these APIs in Elasticsearch needs to provide an async option :)